src/gadgets/nonnative/bn/big_uint.rs

```diff
@@ -74,8 +74,12 @@ impl<F: ff::PrimeField> BigUint<F> {
     pub fn from_limbs(
         limbs: impl Iterator<Item = F>,
         limb_width: NonZeroUsize,
+        limbs_count: NonZeroUsize,
     ) -> Result<Self, Error> {
-        let limbs = limbs.collect::<Vec<_>>();
+        let limbs = limbs
+            .chain(iter::repeat(F::ZERO))
+            .take(limbs_count.get())
+            .collect::<Vec<_>>();
 
         Ok(if limbs.is_empty() {
```

Comment:
with above limbs definition. limbs cannot be empty. we can simply the code here

Reply:
You're right

AND we should to check that the input limbs have not been ignored. I.e., there may be more than takes count